lib/metadata_backfill.rb

```diff
@@ -0,0 +1,26 @@
+require 'stringio'
+
+class MetadataBackfill
+  attr_reader :version
+
+  def initialize(version)
+    @version = version
+  end
+
+  def backfill
+    file = dir.files.get(key) or return
+    pusher = Pusher.new(nil, StringIO.new(file.body))
+    pusher.pull_spec
+    version.update_attributes(metadata: pusher.spec.metadata || {})
```

Comment:
Also, do we want to push a empty `{}` ? or should we just not update the attribute if `pusher.spec.metadata` is nil